src/util/HttpServer/HttpUtils.h

```diff
@@ -73,17 +77,33 @@ static auto createOkResponse(std::string text, const HttpRequest auto& request,
                                       request, mediaType);
 }
 
+static void setBody(http::response<streamable_body>& response,
```

Comment:
Needs a comment what this does (apply compression to generator, create and insert a body, set headers according to the compression.